### ISSUE TITLE
terraform_empty_list_equality rule does not cover non-trivial expressions

### ISSUE THREAD

**Issue by sergei-ivanov** [terraform-linters/tflint#1424]
## Introduction

Thanks for adding this rule, but unfortunately it only picks up the most trivial cases.

For example, it picks up this:
```hcl
resource "aws_security_group_rule" "outbound_proxy" {
  count =var.proxy_subnets != [] ? 1 : 0
```

but not this:
```hcl
resource "aws_security_group_rule" "outbound_proxy" {
  count = var.enabled && var.proxy_subnets != [] ? 1 : 0
```

nor this:
```hcl
templatefile("${path.module}/jenkins.yaml.tmpl", {
    JENKINS_INGRESS_ENABLED = var.jenkins_controller_alb_allow_list != []
})
```

I guess it needs to implement more complex recursion for parsing into expressions.

It may also need to check for comparison with empty maps, e.g. `var.peering_connections != {}`.

## Expected Behavior

All cases of comparison to empty list (or map) are highlighted by the rule.

**Comment by bendrucker** [comment]
The first example would be worth attempting. Short term, the primary goal should be to support expressions commonly passed to attributes like `count`.

Recursive evaluation of complex attribute values is a deeper issue than this rule, e.g. see https://github.com/terraform-linters/tflint/issues/1257. Providing a helper function that visits each value expression within collection types is its own exercise with potential for use in many rules.

**Comment by wata727** [comment]
Fixed by https://github.com/terraform-linters/tflint/pull/1426
